python/mosaic/api/raster.py

```diff
@@ -99,83 +105,131 @@ def rst_boundingbox(raster: ColumnOrName) -> Column:
-    Column (StringType)
-        The path to the clipped raster.
+    Column (RasterTileType)
+        Mosaic raster tile struct column.
 
     """
     return config.mosaic_context.invoke_function(
-        "rst_clip", pyspark_to_java_column(raster), pyspark_to_java_column(geometry)
+        "rst_clip", pyspark_to_java_column(raster_tile), pyspark_to_java_column(geometry)
     )
 
 
-def rst_combineavg(rasters: ColumnOrName) -> Column:
+def rst_combineavg(raster_tiles: ColumnOrName) -> Column:
     """
     Combines the rasters into a single raster.
 
     Parameters
     ----------
-    rasters : Column (ArrayType(StringType))
+    raster_tiles : Column (ArrayType(RasterTileType))
         Raster tiles to combine.
 
     Returns
     -------
-    Column (RasterTile)
+    Column (RasterTileType)
+        The combined raster tile.
+
+    """
+    return config.mosaic_context.invoke_function(
+        "rst_combineavg", pyspark_to_java_column(raster_tiles)
+    )
+
+
+def rst_combineavgagg(raster_tile: ColumnOrName) -> Column:
+    """
+    Combines the aggregate raster tiles into a single tile.
+
+    Parameters
+    ----------
+    raster_tile : Column (RasterTileType)
+        Aggregate raster tile col to combine.
+
+    Returns
+    -------
+    Column (RasterTileType)
         The combined raster tile.
 
     """
     return config.mosaic_context.invoke_function(
-        "rst_combineavg", pyspark_to_java_column(rasters)
+        "rst_combineavgagg", pyspark_to_java_column(raster_tile)
     )
 
 
-def rst_derivedband(raster: ColumnOrName, pythonFunc: ColumnOrName, funcName: ColumnOrName) -> Column:
+def rst_derivedband(raster_tile: ColumnOrName, python_func: ColumnOrName, func_name: ColumnOrName) -> Column:
     """
     Creates a new band by applying the given python function to the input rasters.
     The result is a raster tile.
 
     Parameters
     ----------
-    raster : Column (StringType)
-        Path to the raster file.
-    pythonFunc : Column (StringType)
+    raster_tile : Column (RasterTileType)
+        Mosaic raster tile struct column.
+    python_func : Column (StringType)
         The python function to apply to the bands.
-    funcName : Column (StringType)
+    func_name : Column (StringType)
         The name of the function.
 
     Returns
     -------
-    Column (StringType)
-        The path to the new raster.
+    Column (RasterTileType)
+        Mosaic raster tile struct column.
 
     """
     return config.mosaic_context.invoke_function(
         "rst_derivedband",
-        pyspark_to_java_column(raster),
-        pyspark_to_java_column(pythonFunc),
-        pyspark_to_java_column(funcName),
+        pyspark_to_java_column(raster_tile),
+        pyspark_to_java_column(python_func),
+        pyspark_to_java_column(func_name),
     )
 
 
-def rst_georeference(raster: ColumnOrName) -> Column:
+def rst_derivedbandagg(raster_tile: ColumnOrName, python_func: ColumnOrName, func_name: ColumnOrName) -> Column:
```

Comment:
@mjohns-databricks other agg functions we did separate with _agg.
I am fine with the name either way, just wanted to make sure this aligns.

Reply:
ok, will make the change.